src/main/java/org/fcrepo/importexport/importer/Importer.java

```diff
@@ -648,26 +651,27 @@ public static boolean verifyBag(final File bagDir) {
     /**
      * Method to find and set the repository root from the resource uri.
      * @param uri the URI for the resource
-     * @throws IOException
-     * @throws FcrepoOperationFailedException
      */
-    private void findRepositoryRoot(final URI uri) {
-        repositoryRoot = uri;
+    @VisibleForTesting
+    public URI findRepositoryRoot(final URI uri) {
```

Comment:
I believe all of those `findRepositoryRoot` calls do make HTTP requests.  some of them will get 404 responses, but the `findRepositoryRoot` logic will continue chopping off path elements until it finds the root resource.